lib/valueflow.cpp

```diff
@@ -1551,27 +1589,6 @@ static bool valueFlowForward(Token * const               startToken,
                 }
             }
 
-            // bailout if address of var is taken..
```

Comment:
If I remove the bailout in ```valueFlowInjectParameter``` due to the non-const reference parameter no test object fails.
However, I can construct an example were the behavior without the bailout gives better results, i.e.
```
void dostuff(int & x) {
  f(x);
}
void test() {        
    static int x;
    if (a) x = 2;
    dostuff(x); 
}
```
Without the bailout I get the following debug output:
```
##file test.cpp
1: void dostuff ( int & x@1 ) {
2: f ( x@1 ) ;
3: }
4: void test ( ) {
5: static int x@2 ;
6: if ( a ) { x@2 = 2 ; }
7: dostuff ( x@2 ) ;
8: }



##Value flow
Line 2
  x possible 2
Line 6
  2 always 2
Line 7
  x possible 2
```
With the bailout the value flow information is reduced, i.e. the possible x=2 value on line 2 is missing.
```
##Value flow
Line 6
  2 always 2
Line 7
  x possible 2
```